File: backend/src/main/java/com/worktime/entity/Task.java
package com.worktime.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Table(name = "tasks")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Task {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(length = 1000)
    private String description;

    private LocalDate dueDate;

    @Column
    private Integer estimatedDurationMinutes;

    @Builder.Default
    @Enumerated(EnumType.STRING)
    @Column(name = "priority", nullable = false)
    private TaskPriority priority = TaskPriority.MEDIUM;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private TaskStatus status;

    @ManyToOne
    @JoinColumn(name = "project_id", nullable = false)
    private Project project;

    @ManyToOne
    @JoinColumn(name = "assigned_user_id")
    private User assignedUser;
}

File: frontend/src/app/core/models/task.ts
import { TaskStatus } from './task-status';

export interface Task {
  id: number;
  title: string;
  description: string;
  dueDate: string;
  estimatedDurationMinutes: number;
  priority: string;
  status: TaskStatus;
  projectId: number;
  projectName: string;
  assignedUserId: number | null;
  assignedUserName: string | null;
}